src/watertap_contrib/reflo/core/solar_energy_base.py

```diff
@@ -56,16 +67,90 @@ class SolarEnergyBaseData(UnitModelBlockData):
         ),
     )
     CONFIG.declare(
-        "solar_energy_type",
+        "solar_model_type",
+        ConfigValue(
+            default=SolarModelType.surrogate,
+            domain=In(SolarModelType),
+            description="Solar model type construction flag",
+            doc="""Indicates what type of solar model will be constructed""",
+        ),
+    )
+    CONFIG.declare(
+        "surrogate_model_file",
         ConfigValue(
-            default="PV",
-            domain=In(SolarEnergyType),
-            description="Indicates type of solar energy source",
+            default=None,
+            domain=str,
+            description="Path to surrogate model file",
+            doc="""User provided surrogate model .json file. Must be in same directory as unit model file.""",
```

Comment:
> Must be in same directory as unit model file.

That seems like an undesirable requirement. Is there a reason for this restriction other than this was quickest and most convenient way to implement?

Reply:
Good catch. No I don't think there is any reason this can't be anywhere if user is providing the whole path.